src/jackdaw/test/fixtures.clj

```diff
@@ -111,14 +111,15 @@
     (let [builder (k/streams-builder)
           stream (k/kafka-streams (topology builder) config)
           error (atom nil)
-          started (atom false)]
+          started? (promise)]
 
       (.setUncaughtExceptionHandler stream (set-error error))
-      (.setStateListener stream (set-started (get config "application.id") started))
+      (.setStateListener stream (set-started (get config "application.id") started?))
 
       (k/start stream)
 
-      (when @started)
```

Comment:
This was the bug. At one point (pre-jackdaw), `started` was a promise that blocked until the app under test was ready. When it got changed to an atom, most of the time it would be ok but if the app is slow to reach the "running" state, the test might start firing requests before the app is able to handle them.